### ISSUE TITLE
Create badges.m.o. site map

### ISSUE THREAD

**Issue by emgollie** [mozilla/mozbadging#18]
Need to develop a site map with team feedback per https://github.com/mozilla/mozbadging/issues/17.

**Comment by emgollie** [comment]
Before flushing this out, I want to work with @zzzndr to refine the primary navigation per user feedback (https://github.com/mozilla/mozbadging/issues/27) and initial UI concepts (https://github.com/mozilla/mozbadging/issues/30). I would suggest directing users toward either the earner or issuer flow:
![screen shot 2014-06-23 at 6 49 54 pm](https://cloud.githubusercontent.com/assets/1174853/3366255/ed2a61b4-fb41-11e3-9d95-59be1467fc9e.png)

**Comment by zzzndr** [comment]
I think I would choose to direct users to the earner flow.
Without having hard data to back me up, I think there are more earners then issuers (:

Perhaps it is too time consuming, but what if there will be button on the homepage which toggles the flow from earner to issuer?

**Comment by adamlofting** [comment]
Can we also think about where within this flow we'd like to surface contributor goals and data? (whether that's embedded graphs, or numbers, or just 'find out more' links).

The Mozilla wide contributor dashboard is taking shape here now: http://areweamillionyet.org (see the team drop-down for team specific views).

**Comment by emgollie** [comment]
Thanks @zzzndr and @adamlofting. Per the question "what if there will be button on the homepage which toggles the flow from earner to issuer?" it could be interesting to put together a rough mockup of this toggle interaction to usability test.

For now, we can consider small and medium-size issuers the primary site users but can also validate that with users. Thanks!

**Comment by emgollie** [comment]
Updated site map per discussions with @zzzndr. @adamlofting, per your question I imagine that contributor goals and data could be tracked on the issued and claimed badges dashboard but am open to other thoughts (including whether that become its own page). @christensenep @andrewhayward @davidascher, look forward to your thoughts.   

![sitemap-01](https://cloud.githubusercontent.com/assets/1174853/3465020/090e6b8e-025c-11e4-935d-f25a284a7dc1.jpg)

**Comment by andrewhayward** [comment]
I don't quite understand the "Backpack Mozillians.org display" bit - what does that actually mean?

I'm also not sure how easy it is to separate out the 'earner flow' like this. What _is_ the earner flow? In my mind at least, you've either been given a claim code, which just requires an obvious box somewhere, or you're looking at a badge you want to apply for. There doesn't seem to be much of a distinct set of tasks that make up this flow.

Maybe you just need to explain the site map a little bit for me to actually get it, but I'm somewhat confused by the flow aspects.

**Comment by andrewhayward** [comment]
@zzzndr asked me to briefly explain the new badge creation workflow, for the purposes of the site map:
1. The user is presented with the badge studio.
   
   ![make-a-badge-step-1](https://cloud.githubusercontent.com/assets/159848/3472194/73542df4-02c9-11e4-9661-33525ce4efe1.png)
2. They design their badge.
   
   ![make-a-badge-step-2](https://cloud.githubusercontent.com/assets/159848/3472204/81c4a2ce-02c9-11e4-99c8-3ded2feb1be2.png)
3. When they're happy, they create their badge with the new image.
   
   ![make-a-badge-step-3](https://cloud.githubusercontent.com/assets/159848/3472217/96640850-02c9-11e4-91c0-a5ddddafd3b0.png)
4. Having made their new badge, they are taken to its page on the site.
   
   ![make-a-badge-step-4](https://cloud.githubusercontent.com/assets/159848/3472219/9c69c7c6-02c9-11e4-8fb7-64b235508484.png)

I'm hoping that the studio is obvious enough that it's fairly self-explanatory, but we might need to do some sort of walk-through for users who don't get it.

**Comment by zzzndr** [comment]
thanks, very helpfull!

**Comment by emgollie** [comment]
Updated sitemap per conversations with @zzzndr:
![screen shot 2014-07-03 at 3 02 15 pm](https://cloud.githubusercontent.com/assets/1174853/3476340/bf821392-02fd-11e4-9756-5208fdd2b9b5.png)

**Comment by emgollie** [comment]
Discussed with @davidascher and it seems that we need to create a connection between team/product badges and their available badges. Will keep considering additional connections that aren't currently noted. cc @zzzndr @christensenep

**Comment by zzzndr** [comment]
Here is the revised sitemap where the connections between team & badges are described
@emgollie @davidascher @christensenep 

![sitemap3](https://cloud.githubusercontent.com/assets/6721026/3481604/9d66d942-0374-11e4-9e61-50d5919ed6c9.jpg)

**Comment by emgollie** [comment]
Regarding @andrewhayward's earlier question ("I don't quite understand the 'Backpack Mozillians.org display' bit - what does that actually mean?"), there will need to be connections between an earner's badges and the sites where they are housed and displayed beyond badges.m.o. @christensenep mentioned that he should be getting more information regarding the Mozillians.org display component this week. Once we have those initial details, it would be great discuss ways that we imagine Backpack and Mozillians.org connections taking place (I can imagine simple links but maybe we want to enable something more dynamic).

**Comment by andrewhayward** [comment]
@emgollie - are we to take that as saying that a badges.m.o profile should be a functioning backpack then? Initially I was under the impression that mozillians.org would be the final end-point for badges, but according to [this ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=680209), that's clearly not the case. Unless things have changed in the last year or two, of course :)

As things stand, you need to be a vouched Mozillian to create a badge, but there's nothing to stop you from having an account on badges.m.o without knowing anything about Mozilla. In fact, there's really nothing driving anyone to be further involved in Mozilla at all if and when they receive a badge.

So in some respects I think the biggest split we're looking at (particularly as we're trying to push the number of 'contributors') is not between issuers and earners, but people who are already vouched Mozillians, and those who are not/don't have a Mozillians account at all. How (and if) we surface that is  left as an exercise for the reader.

**Comment by ghost** [comment]
What do you mean by final end-point, exactly?  What I was expecting was that mozillians.org would simply poll badgekit-api for any badges earned by the logged-in user, and then display them on their profile page.  So once you earned a badge, it would just automatically show up there.

Eventually we'd have to add some sort of ability for the user to choose which badges were displayed there (and whether this involves pushing some sort of metadata to badgekit-api, or is managed entirely on mozillians.org is up for debate), but for the most part I was imagining mozillians.org as a "displayer".  It wouldn't house the badges themselves, just display them.

Possibly I'm reading too much into word choices from the comments above, but I wanted to make sure everyone was roughly on the same page as I was here.

**Comment by emgollie** [comment]
@christensenep: You're correct that Mozillians.org would be a display site (sorry to confuse). Thinking through @andrewhayward's larger questions and curious to see where people net out on those.

**Comment by andrewhayward** [comment]
@christensenep: "What do you mean by final end-point, exactly?" I suppose the question I was trying to ask (badly) was 'where are we ultimately trying to direct users to see an individual's badges?' Where is the "backpack" in all of this?

My understanding now is that mozillians.org will just be showing badges from BadgeKit. Is that the case with badges.m.o too? Do Mozillians get a Mozilla-hosted "backpack" anywhere at all?

**Comment by emgollie** [comment]
I see what you're asking, thanks @andrewhayward. In considering the user flow upon claiming a badge, I imagine that users would be directed to backpack.openbadges.org. I think we also want to include some information on badges.mozilla.org encouraging earners to display their badges on their Mozillians.org profiles or create new profiles. Thoughts?

We should also consider badge "favoriting" in this flow (per https://redpen.io/qyf0e54b5a00a98223). I imagine someone can see their saved "favorited" badges when in logging with Persona but that everything related to creating collections of earned badges and displaying earned badges takes place beyond badges.m.o. cc @adamlofting @davidascher @christensenep @zzzndr

**Comment by andrewhayward** [comment]
@emgollie - do we really want to keep using backpack.openbadges.org? Have we discussed having an official Mozilla-hosted equivalent? I suppose I just keep coming back to the idea of increasing contributor numbers, and I'm not really sure what we're doing with badges.mozilla.org to help with that.

**Comment by emgollie** [comment]
@andrewhayward I don't know that a Mozilla-hosted backpack equivalent has been discussed (but @davidascher @christensenep may know). It seems out of scope for this summer but could be beneficial in the long-term as part of a larger federation of backpacks. Eager to hear what others think.

**Comment by ghost** [comment]
I suppose I am honestly not entirely clear on what the benefit of a backpack is/would be if we already have a person's badges being displayed on mozillians.org.  (Not trying to be overly obtuse, as I do understand the theory behind the backpack, just not sure badge adoption is at a place where it's possible to show the value to an end-user yet)

So I suppose I was thinking that we wouldn't really keep using backpack.openbadges.org in the sense that we wouldn't actively push people there (though of course they are welcome to go there and add our badges their backpacks), but that we also wouldn't be replacing it with any other backpack in the near future.  Unless you count mozillians.org profiles as a "backpack" (which, tbh, I think many people would, at least in the sense that I've heard people use the term).

Possibly I am rambling here a bit.

**Comment by emgollie** [comment]
Let's discuss tomorrow @christensenep--I think this is worth discussing live.

**Comment by davidascher** [comment]
backpack.openbadges.org is being used by a bunch of folks.  let's call that
a separate project, out of scope for this summer's work.
